docs/book/src/component-config-tutorial/api-changes.md

```diff
@@ -67,3 +67,41 @@ mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), options)
 With that out of the way, we can get on to defining our new config!
 
 [tutorial-source]: https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/component-config-tutorial/testdata/project
+
+Add below content to this file ->`/config/manager/controller_manager_config.yaml`
+
+```yaml
+apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
+kind: ControllerManagerConfig
+health:
+  healthProbeBindAddress: :8081
+metrics:
+  bindAddress: 127.0.0.1:8080
+webhook:
+  port: 9443
+leaderElection:
+  leaderElect: true
+  resourceName: ecaf1259.tutorial.kubebuilder.io
+# leaderElectionReleaseOnCancel defines if the leader should step down volume
+# when the Manager ends. This requires the binary to immediately end when the
+# Manager is stopped, otherwise, this setting is unsafe. Setting this significantly
+# speeds up voluntary leader transitions as the new leader don't have to wait
+# LeaseDuration time first.
+# In the default scaffold provided, the program ends immediately after
+# the manager stops, so would be fine to enable this option. However,
+# if you are doing or is intended to do any operation such as perform cleanups
+# after the manager stops then its usage might be unsafe.
+# leaderElectionReleaseOnCancel: true
+```
+
+Add below content to this file ->`/config/manager/kustomization.yaml`
+
+```yaml
+generatorOptions:
+  disableNameSuffixHash: true
+
+configMapGenerator:
+- name: manager-config
+  files:
+  - controller_manager_config.yaml
+```
```

Comment:
```suggestion
```

Update the `default/kustomization.yaml` by adding under the `patchesStrategicMerge:` the following patch:

```yaml

# Mount the controller config file for loading manager configurations
# through a ComponentConfig type
- manager_config_patch.yaml
```

```

```